wave_controls/src/pid.cpp

```diff
@@ -0,0 +1,51 @@
+#include "wave/controls/pid.hpp"
+
+namespace wave {
+
+PID::PID(void) {
+    this->error_prev = 0.0;
+    this->error_sum = 0.0;
+
+    this->error_p = 0.0;
+    this->error_i = 0.0;
+    this->error_d = 0.0;
+
+    this->k_p = 0.0;
+    this->k_i = 0.0;
+    this->k_d = 0.0;
+}
+
+PID::PID(double k_p, double k_i, double k_d) {
+    this->error_prev = 0.0;
+    this->error_sum = 0.0;
+
+    this->error_p = 0.0;
+    this->error_i = 0.0;
+    this->error_d = 0.0;
+
+    this->k_p = k_p;
+    this->k_i = k_i;
+    this->k_d = k_d;
+}
+
+double PID::update(double setpoint, double actual, double dt) {
+    double error;
```

Comment:
I do. If anything AAA makes it easier to code without accidentally uninitialized variables and subtle conversion bugs. Meyers addresses this in _Effective Modern C++_ ("Item 5: Prefer auto to explicit type declarations") , with some examples.

> The fact of the matter is that writing types explicitly often does little more than intro‐
duce opportunities for subtle errors, either in correctness or efficiency or both.

Herb Sutter also has a good response to the "readability" argument here:
https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

There are cases where you are right to be concerned and naive `auto` use is not good, in particular with Eigen expressions. (see _Effective Modern C++_ Item 6.) AAA can still be helpful here, but it does require further discussion.